pagerduty/resource_pagerduty_service_dependency.go

```diff
@@ -160,26 +160,30 @@ func resourcePagerDutyServiceDependencyDisassociate(d *schema.ResourceData, meta
 
 	log.Printf("[INFO] Disassociating PagerDuty dependency %s", dependency.DependentService.ID)
 
-	// listServiceRelationships by calling get dependencies using the serviceDependency.DependentService.ID
-	depResp, _, err := client.ServiceDependencies.GetServiceDependenciesForType(dependency.DependentService.ID, dependency.DependentService.Type)
-	if err != nil {
-		return err
-	}
-
 	var foundDep *pagerduty.ServiceDependency
 
-	// loop serviceRelationships until relationship.IDs match
-	for _, rel := range depResp.Relationships {
-		if rel.ID == d.Id() {
-			foundDep = rel
-			break
+	// listServiceRelationships by calling get dependencies using the serviceDependency.DependentService.ID
+	retryErr := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		if dependencies, _, err := client.ServiceDependencies.GetServiceDependenciesForType(dependency.DependentService.ID, dependency.DependentService.Type); err != nil {
+			if isErrCode(err, 404) || isErrCode(err, 500) || isErrCode(err, 429) {
```

Comment:
@stmcallister, would be a good idea to centralize the logic of what is a returnable error? I'm not sure if it is worth it to retry on 400 at all for example. I can see the benefit of increasing the retry to 5min for 429 and maybe 503/504 errors, but it might substantially hurt the experience if there is a real 404 or 400.

Reply:
@drastawi Thanks for the feedback! The tricky part is that in some race condition situations we do want to retry on a 404 🤔 Good point on the centralizing of the logic.